qiskit_ibm_provider/utils/utils.py

```diff
@@ -201,6 +202,15 @@ def _filter_value(
                 _filter_value(value, filter_keys)
 
 
+def are_circuits_dynamic(circuits: Union[QuantumCircuit, List[QuantumCircuit]]) -> bool:
+    """Checks if the input circuits are dynamic."""
+    for circuit in circuits:
+        for inst in circuit:
+            if isinstance(inst, ControlFlowOp):
+                return True
```

Comment:
I think you're missing

```
        if getattr(inst.operation, "condition", None) is not None:
            return True
```

Reply:
I believe this is unnecessary, because all control instructions inherit from `ControlFlowOp` :
```
class ControlFlowOp(Instruction, ABC):
    """Abstract class to encapsulate all control flow operations."" 
```

Reply:
That was a suggestion from @jakelishman, so maybe you can follow up with him on whether that's needed. I thought that'd cover the case of a custom instruction that's not a `ControlFlowOp`, but some quick testing showed that's not true.

Reply:
The `condition` field is the old form for specifying single instructions that are conditioned on a classical value.  The IBM Provider backends inject a custom transpiler pass into Terra's pipelines to convert any instruction with a non-`None` `condition` field to a new-style `IfElseOp`, so you ideally shouldn't see any cases where that logic triggers if the user has correctly transpiled their circuit for your backend.

That check is necessary in the general case, though; it's still (currently) a valid way to define a conditional instruction in Terra.